### PR TITLE
Update to Feed Source deployed feed version

### PR DESCRIPTION
**Pull request by br648**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Update to how a feed source's feed version is obtained. This now takes into consideration a pinned feed version.

### CODE REVIEW

**Comment by miles-grant-ibigroup** [comment]
I will be second reviewer on this

**Review by philip-cline — APPROVED** [review]
This looks good for me, I'm seeing the same behaviour as before the update.

File: src/test/java/com/conveyal/datatools/manager/controllers/api/FeedSourceControllerTest.java [hunk 1]
```diff
@@ -255,20 +276,10 @@ public void createFeedSourceWithLabels() throws IOException {
...
-        DeployJob deployJob = new DeployJob(
-            "Deploying " + deploymentDeployed.name,
-            deploymentDeployed,
-            user,
-            new OtpServer(),
-            "test-deploy",
-            DeployJob.DeployType.USE_PRELOADED_BUNDLE,
-            true
-        );
-        deployJob.run();
```
Comment (miles-grant-ibigroup) [inline src/test/java/com/conveyal/datatools/manager/controllers/api/FeedSourceControllerTest.java:271]: why was this removed?

File: src/main/java/com/conveyal/datatools/manager/models/FeedSource.java [hunk 2]
```diff
@@ -479,77 +459,101 @@ public String latestVersionId() {
...
-    private FeedVersion getDeployedFeedVersionViaExpensiveQuery() {
+    private FeedVersion getDeployedFeedVersion() {
         if (deployedFeedVersion == null) {
-            Collection<Deployment> deployments = Persistence.deployments.getFiltered(eq("projectId", this.projectId), Sorts.descending("lastUpdated"));
-            Collection<FeedVersion> feedVersions = Persistence.feedVersions.getFiltered(eq("feedSourceId", this.id), Sorts.descending("updated"));
-            if (deployments.isEmpty() || feedVersions.isEmpty()) {
-                return null;
+            Project project = Persistence.projects.getById(this.projectId);
+            if (project.pinnedDeploymentId != null) {
+                Deployment deployment = Persistence.deployments.getById(project.pinnedDeploymentId);
```
Comment (miles-grant-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/models/FeedSource.java]: I believe this line is causing the slowdown. Would it be possible to cache the deployments in memory on startup and try searching that cache before resorting to this?

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Works really well and behavior is correct! However, the slowdown is significant, and I think that should be addressed before we merge

**Comment by philip-cline** [comment]
Also thinking out loud here, but could we potentially keep the old code but only update the mongo property when a pinned deployment is deployed? Or we still have a separate issue with other information being updated there?

File: src/test/java/com/conveyal/datatools/manager/controllers/api/FeedSourceControllerTest.java (hunk 1 above)
Comment (br648) [reply src/test/java/com/conveyal/datatools/manager/controllers/api/FeedSourceControllerTest.java:271]: It's not needed because I also removed the code related to this task from the DeployJob. Running this job no longer has and effect on defining the deployed feed version.

File: src/main/java/com/conveyal/datatools/manager/models/FeedSource.java (hunk 2 above)
Comment (br648) [reply src/main/java/com/conveyal/datatools/manager/models/FeedSource.java]: I have made some further updates to reduce objects being returned and in the case of deployment objects limiting the fields returned. I'm hoping this is enough to avoid caching as managing that will be quite onerous.

File: src/main/java/com/conveyal/datatools/manager/persistence/TypedPersistence.java [hunk 3]
```diff
@@ -176,11 +188,20 @@ public List<T> getFiltered (Bson filter) {
...
-        }
+        return (sortBy != null)
+        ? mongoCollection.find(filter).sort(sortBy).into(new ArrayList<>())
+        : mongoCollection.find(filter).into(new ArrayList<>());
+    }
+
+    /**
+     * Get all objects populating only the included fields satisfying the supplied Mongo filter and sort by.
+     */
+    public List<T> getFilteredLimitedFields(Bson filter, Bson sortBy, String... includedFields) {
```
Comment (miles-grant-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/persistence/TypedPersistence.java:199]: It's better than before but it's still quite slow... 

Is there any way to make the request only once on startup and then hold it? that should be less extreme than fully caching the results. we could fallback to making this request if we don't find the data in that initial request data.

I tried fiddling with mongo settings but couldn't get performance better enough

**Review by miles-grant-ibigroup — APPROVED** [review]
Going to approve this so this is fixed, but think we should still attempt the initial request method if possible! If you want to discuss more outside of GitHub let me know

**Comment by miles-grant-ibigroup** [comment]
Let's merge this, then move forward with our approach of moving the expensive fields to a new endpoint
